### PR TITLE
Add intro to Program Setup

### PR DESCRIPTION
**Pull request by gpsimenos**

### CODE REVIEW

File: docs/program-setup/runtime/Execution.md
Comment (iriark01) [inline docs/program-setup/runtime/Execution.md]:
```suggestion
    * You won't usually need a serial driver for Mbed enabled boards, but if you're on Windows you [may need to install one](../program-setup/windows-serial-driver.html).
```

File: docs/program-setup/runtime/Execution.md
Comment (iriark01) [inline docs/program-setup/runtime/Execution.md]: I'll move this to a new file when we're done with the text review

File: docs/program-setup/runtime/Execution.md
Comment (iriark01) [inline docs/program-setup/runtime/Execution.md]:
```suggestion
* Set up a serial connection to your board so you can flash and debug your program.
```

File: docs/program-setup/runtime/Execution.md
Comment (iriark01) [inline docs/program-setup/runtime/Execution.md]: What does that mean? That I don't need to install a terminal app?

Comment (iriark01) [inline docs/program-setup/runtime/Execution.md]:
```suggestion
    * Mbed CLI has a built-in serial terminal to display output from your program.
```

Comment (iriark01) [reply docs/program-setup/runtime/Execution.md]: Just so it matches "flash and debug your program" - you can also change both to application, which is far more common in Mbed OS ("program" is being dragged over from Mbed Studio)

File: docs/program-setup/runtime/Execution.md
Comment (iriark01) [inline docs/program-setup/runtime/Execution.md]: Do these need more info? How can we help the user decide which topic to read when?

File: docs/program-setup/runtime/Execution.md
Comment (iriark01) [inline docs/program-setup/runtime/Execution.md]: I made this comment on the previous PR - I'm not sure how the first paragraph relates to the rest of the topic, and why the full/bare metal comparison repeats in both paragraphs.

File: docs/program-setup/runtime/Execution.md
Comment (iriark01) [inline docs/program-setup/runtime/Execution.md]: This is a repeat of the previous paragraph - which one did you want to keep?

File: docs/program-setup/runtime/Execution.md
Comment (iriark01) [inline docs/program-setup/runtime/Execution.md]: Move the one we don't support to be the second, I think, and just call it

* Cooperative (not supported in Mbed OS): The active thread needs to yield execution for another thread to run.

File: docs/program-setup/runtime/Execution.md
Comment (iriark01) [inline docs/program-setup/runtime/Execution.md]:
```suggestion
* Preemptive: A scheduler (a central piece of software) is responsible for piking and running the threads. Schedulers can use different algorithms: 

    * Priority based round robin (used in Mbed OS).
    * Execute threads in equal time slots, according to their priority.    
```

File: docs/program-setup/runtime/Execution.md
Comment (iriark01) [inline docs/program-setup/runtime/Execution.md]: So this may be my confusion, but it seems that we say "there are two types of schedulers: a. cooperative b. a scheduler"

**Review by iriark01 — CHANGES_REQUESTED** [review]

File: docs/program-setup/runtime/Execution.md
Comment (gpsimenos) [reply docs/program-setup/runtime/Execution.md]: Yes you can use ```mbed sterm``` as described further in the docs.

File: docs/program-setup/runtime/Execution.md
Comment (gpsimenos) [reply docs/program-setup/runtime/Execution.md]: Removed the duplicate comparison. I incluced this information because Evelyne asked me to provide an intro to embedded systems/embedded programming. If not relevant, I can remove it.

File: docs/program-setup/runtime/Execution.md
Comment (gpsimenos) [reply docs/program-setup/runtime/Execution.md]: Ooops, removed the duplicate.

File: docs/program-setup/runtime/Execution.md
Comment (gpsimenos) [reply docs/program-setup/runtime/Execution.md]: Clarified.

File: docs/program-setup/runtime/Execution.md
Comment (gpsimenos) [reply docs/program-setup/runtime/Execution.md]: I'm not sure more info would be beneficial here. These topics are well documeneted in their respective pages. Unless you mean definitions of the terms like "build profile", "target" etc?

File: docs/program-setup/runtime/Execution.md
Comment (iriark01) [reply docs/program-setup/runtime/Execution.md]: But still, I don't see how the content of this paragraph relates to the rest of this page.

Comment (gpsimenos) [reply docs/program-setup/runtime/Execution.md]: Removed

**Review by rajkan01 — APPROVED** [review]
LGTM

**Comment by LDong-Arm** [comment]
The change in this PR looks good to me, but it needs a manual rebase because files have been moved.

The current structure of `Execution.md` looks slightly weird to me - it starts with a section to explain threads, then another section to say Mbed OS has two modes - thread and handler, and a sub-section to explain the handler mode. Though this discussion is probably outside this PR's scope as the page was like this before.

**Comment by iriark01** [comment]
I think that's a fair point, though, and if we're already on this page we may want to do the whole thing at once. However, that is up to @gpsimenos , as it goes beyond the scope of his original task.

**Comment by gpsimenos** [comment]
Rebased. Let's open a separate PR after this is merged to reconsider the layout of the page.

**Review by LDong-Arm — APPROVED** [review]
LGTM on the technical side

**Comment by iriark01** [comment]
In any new PR you open, I would like to understand how this page relates to 
https://os.mbed.com/docs/mbed-os/v6.0/apis/scheduling-concepts.html
https://os.mbed.com/docs/mbed-os/v6.0/apis/scheduling-tutorials.html

I think one reason Evelyne may have wanted more details on this page is that the info may be elsewhere - and is neither in the same area of the docs nor linked.

**Comment by iriark01** [comment]
Thank you for the content! It'll be published with the upcoming release

**Pull request merged**
